src/screens/Login/index.tsx

```diff
@@ -1,15 +1,123 @@
+        await SecureStore.setItemAsync('refreshToken', refreshToken);
+        setErrorMessage('') // Clear error message
+        navigation.navigate('Home'); // Navigate to the home screen on success
+      }
+    } catch (error) {
+      console.log(error,'rrorrrr');
+      if (error && error.message) {
+        setErrorMessage(error.message); // Set the specific error message for display
+      } else {
+        setErrorMessage('An error occurred. Please try again.'); // Fallback error message
+      }
+    }
+  };
+  const handleRegister = () => {
+    navigation.navigate('Register'); // Replace 'Register' with the name of your registration screen or route
+  };
+  const isFormValid = !errors.email && !errors.password; // Check if both email and password fields have no errors
 
-function Login() {
   return (
-    // <Column safeAreaTop mt={30} h="100%">
-    <Center safeAreaTop mt={30} h="100%">
-      <Heading textAlign="center" mb={6}>
-        Welcome to CraftPile!
-      </Heading>
-    </Center>
-    // </Column>
+      <Center flex={1} py="10">
+        <Box w="90%">
+          <Heading size="lg" fontWeight="600" color="coolGray.800" _dark={{ color: 'warmGray.50' }}>
+            Welcome to CraftPile!
+          </Heading>
+          <Heading mt="1" _dark={{ color: 'warmGray.200' }} color="coolGray.600" fontWeight="medium" size="xs">
+            Sign in to continue!
+          </Heading>
+          <VStack space={3} mt="5">
+            <FormControl isInvalid={!!errors.email?.message}>
+              <FormControl.Label>Email ID</FormControl.Label>
+              <Controller
+                control={control}
+                name="email"
+                render={({ field: { onChange, onBlur, value } }) => (
+                  <Input autoCapitalize="none" onChangeText={onChange} onBlur={onBlur} value={value} />
+                )}
+                rules={{
+                  required: { value: true, message: 'Email is required' },
+                  pattern: { value: emailPattern, message: 'Invalid email address' },
+                }}
+              />
+              {errors.email && <FormControl.ErrorMessage>{errors.email.message}</FormControl.ErrorMessage>}
+            </FormControl>
+            <FormControl isInvalid={!!errors.password?.message}>
+              <FormControl.Label>Password</FormControl.Label>
+              <Controller
+                control={control}
+                name="password"
+                render={({ field: { onChange, onBlur, value } }) => (
+                  <Input onChangeText={onChange} onBlur={onBlur} value={value} secureTextEntry />
+                )}
+                rules={{
+                  required: { value: true, message: 'Password is required' },
+                  pattern: {
+                    value: passwordPattern,
+                    message: 'Password must contain at least 8 characters, including at least one uppercase letter, one lowercase letter, and one number',
```

Comment:
We don't need to validate the password input on Login since we already did in Registration. All we really care about here is that the field is not blank.